packages/eslint-plugin-baseui/__tests__/deprecated-component-api.test.js

```diff
@@ -475,6 +475,49 @@ const tests = {
       `,
     },
 
+    // oldName and newName imported and used
+    {
+      code: `
+        import { HeadingXXLarge, H1 } from "baseui/typography"
+
+        const MarkdownRender = () => {
+          return <div><H1>Large</H1><HeadingXXLarge>Large</HeadingXXLarge></div>
+          }
+      `,
+      errors: [
+        {messageId: MESSAGES.remove.id},
+        {messageId: MESSAGES.replace.id},
+      ],
+      output: `
+        import { HeadingXXLarge, H1 } from "baseui/typography"
+
+        const MarkdownRender = () => {
+          return <div><HeadingXXLarge>Large</HeadingXXLarge><HeadingXXLarge>Large</HeadingXXLarge></div>
+          }
+      `,
+    },
+    // oldName imported and renamed, and newName imported and used
+    {
+      code: `
+        import { HeadingXXLarge as Hello, H1 } from "baseui/typography"
```

Comment:
I believe something like this may avoid the need to manually remove the import 
https://astexplorer.net/#/gist/06f1acbd8bc8feb3b6c28f5bc656d579/f99268da03756b87c48c556eaf38c8a429912a06

Reply:
Rewriting the entire import statement using the current node state causes other import changes to be discarded. I've updated the rule to prune from the end of the previous specifier to the end of the current specifier which fixes the issue.